Entities/SceneLayer.cpp

```diff
@@ -163,68 +183,81 @@ namespace RTE {
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-	int SceneLayer::LoadData() {
+	template <bool TRACK_DRAWINGS>
+	int SceneLayerImpl<TRACK_DRAWINGS>::LoadData() {
 		// Load from disk and take ownership. Don't cache because the bitmap will be modified.
 		m_MainBitmap = m_BitmapFile.GetAsBitmap(COLORCONV_NONE, false);
 		m_MainBitmapOwned = true;
+
+		m_BackBitmap = create_bitmap_ex(bitmap_color_depth(m_MainBitmap), m_MainBitmap->w, m_MainBitmap->h);
+		m_LastClearColor = ColorKeys::g_InvalidColor;
+
 		InitScrollRatios();
 		return 0;
 	}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-	int SceneLayer::SaveData(const std::string &bitmapPath) {
+	template <bool TRACK_DRAWINGS>
+	int SceneLayerImpl<TRACK_DRAWINGS>::SaveData(const std::string &bitmapPath) {
 		if (bitmapPath.empty()) {
 			return -1;
 		}
 		if (m_MainBitmap) {
 			// Make a copy of the bitmap to pass to the thread because the bitmap may be offloaded mid thread and everything will be on fire.
-			BITMAP *outputBitmap = create_bitmap_ex(8, m_MainBitmap->w, m_MainBitmap->h);
+			BITMAP *outputBitmap = create_bitmap_ex(bitmap_color_depth(m_MainBitmap), m_MainBitmap->w, m_MainBitmap->h);
 			blit(m_MainBitmap, outputBitmap, 0, 0, 0, 0, m_MainBitmap->w, m_MainBitmap->h);
 
 			auto saveLayerBitmap = [bitmapPath](BITMAP *bitmapToSave) {
 				PALETTE palette;
 				get_palette(palette);
 				if (save_png(bitmapPath.c_str(), bitmapToSave, palette) != 0) {
 					// TODO: This will not kill the main thread. Figure this out!
-					RTEAbort(std::string("Failed to save SceneLayer bitmap to path and name: " + bitmapPath));
+					RTEAbort(std::string("Failed to save SceneLayerImpl bitmap to path and name: " + bitmapPath));
 				}
 				destroy_bitmap(bitmapToSave);
 			};
 			std::thread saveThread(saveLayerBitmap, outputBitmap);
 			m_BitmapFile.SetDataPath(bitmapPath);
-			// TODO: Move this into some global thread container or a ThreadMan� instead of detaching.
+			// TODO: Move this into some global thread container or a ThreadMan� instead of detaching.
```

Comment:
I don't know what happened here. I don't even want to fix this, as a cautionary tale.